example/event-listener/src/index.ts

```diff
@@ -0,0 +1,147 @@
+
+async function main(): Promise<boolean> {
+  const [verifierNode1] = paladin.getVerifiers("member@node1");
+
+  // Create a privacy group for Node1 alone
+  logger.log("Creating a privacy group for Node1...");
+  const penteFactory = new PenteFactory(paladin, "pente");
+  const memberPrivacyGroup = await penteFactory.newPrivacyGroup({
+    members: [verifierNode1],
+    evmVersion: "shanghai",
+    externalCallsEnabled: true,
+  });
+  if (!checkDeploy(memberPrivacyGroup)) return false;
+
+  // Deploy a smart contract within the privacy group
+  logger.log("Deploying a smart contract to the privacy group...");
+  const contractAddress = await memberPrivacyGroup.deploy({
+    abi: helloWorldJson.abi,
+    bytecode: helloWorldJson.bytecode,
+    from: verifierNode1.lookup,
+  });
+  if (!contractAddress) {
+    logger.error("Failed to deploy the contract. No address returned.");
+    return false;
+  }
+  logger.log(`Contract deployed successfully! Address: ${contractAddress}`);
+
+  // Check the latest sequence received
+  const receipts = await paladin.queryTransactionReceipts({
+    limit: 1,
+    sort: ["-sequence"],
+  });
+  const lastSequence = receipts[0].sequence;
+  logger.log(`Last sequence received: ${lastSequence}`);
+
+  // Create a new listener (deleting one if it already exists)
+  logger.log("Creating a receipt listener...");
+  try {
+    await paladin.deleteReceiptListener("example-event-listener");
+  } catch (err) {
+    // do nothing
+  }
+  await paladin.createReceiptListener({
+    name: "example-event-listener",
+    filters: {
+      type: TransactionType.PRIVATE,
+      domain: "pente",
+      sequenceAbove: lastSequence,
+    },
+    options: {
+      domainReceipts: true,
+      incompleteStateReceiptBehavior: "block_contract",
+    },
+  });
+
+  const name = nanoid(10);
+  let received = false;
+
+  // Create a websocket client for node1
+  const processReceipt = async (receipt: ITransactionReceipt) => {
+    if (receipt.domain === undefined) {
+      return;
+    }
+    const domainReceipt = await paladin.getDomainReceipt(
+      receipt.domain,
+      receipt.id
+    );
+    if (domainReceipt !== undefined && "receipt" in domainReceipt) {
+      for (const log of domainReceipt.receipt.logs ?? []) {
+        const decoded = await paladin.decodeEvent(log.topics, log.data);
+        const message = decoded?.data?.message;
+        if (message?.indexOf(name) !== -1) {
+          logger.log(`Received event data: ${JSON.stringify(decoded?.data)}`);
+          received = true;
+        }
+      }
+    }
+  };
+  const wsClient = new PaladinWebSocketClient(
+    {
+      url: "ws://127.0.0.1:31549",
+      subscriptions: ["example-event-listener"],
+    },
+    async (sender, event) => {
+      if (event.method === "ptx_subscription") {
+        for (const receipt of event.params.result.receipts) {
+          await processReceipt(receipt);
+          sender.ack(event.params.subscription);
```

Comment:
Maybe worth a comment on the importance of this?

Reply:
Yes, and I've actually put it in the wrong place... should be outside the loop.